serialization/ascii/unmarshal_utils.go

```diff
@@ -0,0 +1,176 @@
+package ascii
+
+import (
+	"fmt"
+	"reflect"
+)
+
+func errInvalidData(p []byte) error {
+	for i := range p {
+		if p[i] > 127 {
+			return fmt.Errorf("failed to unmarshal ascii: invalid charester %s", string(p[i]))
+		}
+	}
+	return nil
+}
+
+func errNilReference(v interface{}) error {
+	return fmt.Errorf("failed to unmarshal ascii: can not unmarshal into nil reference(%T)(%[1]v)", v)
+}
+
+func DecString(p []byte, v *string) error {
+	if v == nil {
+		return errNilReference(v)
+	}
+	*v = decString(p)
+	return errInvalidData(p)
+}
+
+func DecStringR(p []byte, v **string) error {
+	if v == nil {
+		return errNilReference(v)
+	}
+	*v = decStringR(p)
+	return errInvalidData(p)
+}
+
+func DecBytes(p []byte, v *[]byte) error {
+	if v == nil {
+		return errNilReference(v)
+	}
+	*v = decBytes(p)
+	return errInvalidData(p)
+}
+
+func DecBytesR(p []byte, v **[]byte) error {
+	if v == nil {
+		return errNilReference(v)
+	}
+	*v = decBytesR(p)
+	return errInvalidData(p)
+}
+
+func DecInterface(p []byte, v *interface{}) error {
+	if v == nil {
+		return errNilReference(v)
+	}
+	*v = decBytes(p)
+	return errInvalidData(p)
+}
+
+func DecReflect(p []byte, v reflect.Value) error {
+	if v.IsNil() {
+		return errNilReference(v)
+	}
+
+	switch v = v.Elem(); v.Kind() {
+	case reflect.String:
+		v.SetString(decString(p))
+	case reflect.Slice:
+		if v.Type().Elem().Kind() != reflect.Uint8 {
+			return fmt.Errorf("failed to marshal ascii: unsupported value type (%T)(%[1]v)", v.Interface())
+		}
+		v.SetBytes(decBytes(p))
+	case reflect.Interface:
+		v.Elem().Set(reflect.ValueOf(decBytes(p)))
```

Comment:
I am pretty sure that ascii, by default, should be serialized into `string` not into `[]byte`

Reply:
`text`, `blob`, `varchar` `cql types` unmarshalls `*interface{}` as `[]byte`

Reply:
I don't see any logic in old marshal that would inmarshal to `interface`, but here is the default behavior when it prepares rows: 
```
func goType(t TypeInfo) (reflect.Type, error) {
	switch t.Type() {
	case TypeVarchar, TypeAscii, TypeInet, TypeText:
		return reflect.TypeOf(*new(string)), nil
```

Reply:
it was not supported before, I am pretty sure we don't need to support unmarshaling into `*interface{}`. let's remove it for now.

Reply:
Support unmarshaling into `*interface{}` and `custom *interface{}` removed.

Reply:
thanks